.github/workflows/main.yaml

```diff
@@ -0,0 +1,133 @@
+name: Build and Push Docker images
```

Comment:
As we discussed in the issue, we would like to keep separate pipeline execution for backend and frontend changes, depending on whether they have changed. Can we please implement that?

Reply:
Implemented, please check